src/Stratis.Bitcoin.Features.Consensus/Extensions.cs

```diff
@@ -12,10 +12,8 @@ public static ChainedBlock FindFork(this ChainedBlock newTip, ChainedBlock tip)
         {
             ChainedBlock highest = newTip.Height > tip.Height ? newTip : tip;
             ChainedBlock lowest = highest == newTip ? tip : newTip;
-            while (lowest.Height != highest.Height)
-            {
-                highest = highest.Previous;
-            }
+
+            highest = highest.GetAncestor(lowest.Height);
 
             while (lowest.HashBlock != highest.HashBlock)
```

Comment:
and finally, also not for this PR :-), this loop can be improved by checking if Skip hashes match or not
if i understand it well, both headers should have same height of their skip headers, so comparing hashes of skip headers makes sense and if they are not equal, we can jump to skip instead of to previous

Reply:
I don't think that would work with hashes. What if the matching hash was in between the current block and the skip. Wouldn't you miss it then?

Reply:
yes, but i'm talking about the opposite scenario
if those hashes are different for skips, you can advanced there
but not vice versa

Reply:
Ah ok, yeah I get it now. Yes that will work if the skip heights are the same. Cool idea but too bad this method isn't used :grin: Maybe we could do the same on one of the other FindFork methods?

Reply:
definitely